spiped/dispatch.c

```diff
@@ -80,7 +82,7 @@ doaccept(struct accept_state * A)
 	int rc = 0;
 
 	/* If we can, accept a new connection. */
-	if ((A->nconn < A->nconn_max) && (A->accept_cookie == NULL)) {
+	if ((A->nconn < A->nconn_max) && (A->accept_cookie == NULL) && !(A->shutdown)) {
```

Comment:
This does not work like I expect yet. This change should prevent accepting new connections once `SIGTERM` was received, can you tell me what I did wrong?